doc/adding_new_cmd.md

```diff
@@ -0,0 +1,114 @@
+# Adding a new cli command
```

Comment:
Bit nitpicky but it would be good to standardise on the capitalisation of Mobile CLI/mobile cli/mobile CLI.